core/src/main/java/aima/core/search/api/BidirectionalProblem.java

```diff
@@ -0,0 +1,20 @@
+package aima.core.search.api;
+
+/**
+ * @author manthan
+ */
+public interface BidirectionalProblem<A, S> extends
+        // S initialState()
+        InitialStateFunction<S>,
+        // List<A> actions(S s)
+        ActionsFunction<A, S>,
+        // List<S> results(S s, A a)
+        ResultFunction<A, S>,
+        // boolean isGoalState(S s)
+        GoalTestPredicate<S>,
+        // double stepCost(S s, A a, S s')
+        StepCostFunction<A, S> {
+    Problem<A,S> getOriginalProblem();
```

Comment:
Why are you doing the above? Ideally, I'd like to keep this the way it was. I would like to also avoid changing the NodeFactory to take this into account could you explain further?